lua/nvim-treesitter/indent.lua

```diff
@@ -114,31 +132,36 @@ function M.get_indent(lnum)
     end
 
     -- do not indent for nodes that starts-and-ends on same line and starts on target line (lnum)
-    if not is_processed_by_row[srow] and (q.indent[node:id()] and srow ~= erow and srow ~= lnum - 1) then
+    if
+      not is_processed_by_row[srow]
+      -- Dear stylua, please don't change the semantics of this statement!
+      -- stylua: ignore start
+      and (q.indent[node:id()] and srow ~= erow and ((srow ~= lnum - 1) or q.indent[node:id()].start_at_same_line))
+      -- stylua: ignore end
+    then
       indent = indent + indent_size
       is_processed = true
     end
 
-    if q.aligned_indent[node:id()] and srow ~= erow then
+    -- do not indent for nodes that starts-and-ends on same line and starts on target line (lnum)
+    if q.aligned_indent[node:id()] and srow ~= erow and (srow ~= lnum - 1) then
       local metadata = q.aligned_indent[node:id()]
-      local opening_delimiter = metadata.delimiter:sub(1, 1)
-      local o_delim_node, pos = get_matching_prev_sibling(node, { srow, #vim.fn.getline(srow + 1) - 1 }, function(n)
-        return n:type() == opening_delimiter
-      end)
+      local o_delim_node, is_last_in_line
+      if metadata.delimiter then
+        local opening_delimiter = metadata.delimiter and metadata.delimiter:sub(1, 1)
+        o_delim_node, is_last_in_line = find_delimiter(bufnr, node, opening_delimiter)
+      else
+        o_delim_node = node
+      end
 
       if o_delim_node then
-        if pos == 1 then
+        if is_last_in_line then
           -- hanging indent (previous line ended with starting delimiter)
           indent = indent + indent_size * 1
         else
           local _, o_scol = o_delim_node:start()
-          local aligned_indent = math.max(indent, 0) + o_scol
-          if indent > 0 then
-            indent = aligned_indent
-          else
-            indent = aligned_indent + 1 -- extra space for starting delimiter
-          end
-          is_processed = true
+          o_scol = o_scol + (metadata.increment or 1)
+          return math.max(indent, 0) + o_scol
```

Comment:
avoid modifying the parameter maybe ?
```suggestion
          return math.max(indent, 0) + o_scol + (metadata.increment or 1)
```